File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/SpawnGlobalEntityPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.api.network.packet.Packet;

import io.netty.buffer.ByteBuf;

public final record SpawnGlobalEntityPacket(int entityId, byte type, int x, int y, int z) implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        writeVarInt(os, entityId);
        os.writeByte(type);
        os.writeInt(x);
        os.writeInt(y);
        os.writeInt(z);
    }

    @Override
    public byte getId() {
        return 0x2C;
    }
}

File: server/src/main/kotlin/gg/mineral/server/network/packet/play/clientbound/SpawnGlobalEntityPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class SpawnGlobalEntityPacket(val entityId: Int, val type: Byte, val x: Int, val y: Int, val z: Int) :
    Packet.Outgoing {
    override fun serialize(os: ByteBuf) {
        os.writeVarInt(entityId)
        os.writeByte(type.toInt())
        os.writeInt(x, y, z)
    }

    override val id: Byte
        get() = 0x2C
}
